include/xtensor/xcontainer.hpp

```diff
@@ -358,8 +361,57 @@ namespace xt
         adapt_strides();
         data().resize(data_size());
     }
+
+    /**
+     * Transposes the container inplace by reversing the dimensions.
+     */
+    template <class D>
+    inline void xcontainer<D>::transpose()
+    {
+        // reverse stride and shape
+        std::reverse(m_strides.begin(), m_strides.end());
+        std::reverse(m_shape.begin(), m_shape.end());
+        adapt_strides();
+    }
+
+    /**
+     * Transposes the container inplace by permuting the shape with @permute.
+     * @param permute the dimensions with the given permutation
+     */
+    template <class D>
+    inline void xcontainer<D>::transpose(const std::vector<size_type>& permute)
+    {
+        if (permute.size() != m_shape.size())
+            throw transpose_error(std::string("Permutation does not have the same size as shape"));
+
+        // check if axis is twice in permute
+        for (size_type i = 0; i < permute.size(); ++i)
+            for (size_type j = i + 1; j < permute.size(); ++j)
+                if (permute[i] == permute[j])
+                    throw transpose_error(std::string("Permutation contains axis more than once"));
```

Comment:
A check_policy tag would be more flexible since you may want to check your permutation vector even in "release" mode, if this vector is computed or read from a file for instance. So both behaviors could coexist in a same code.